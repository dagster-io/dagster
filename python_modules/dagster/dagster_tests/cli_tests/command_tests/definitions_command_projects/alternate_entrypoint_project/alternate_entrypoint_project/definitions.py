from dagster import Definitions, asset, define_asset_job


@asset
def my_asset() -> None: ...


my_job = define_asset_job(name="my_job", selection="my_asset")

defs = Definitions(assets=[my_asset], jobs=[my_job])
