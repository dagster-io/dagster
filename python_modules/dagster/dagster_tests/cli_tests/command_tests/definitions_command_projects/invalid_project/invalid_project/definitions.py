import dagster as dg


@dg.asset(name="my_asset")
def my_first_asset() -> None: ...


@dg.asset(name="my_asset")
def my_second_asset() -> None: ...


my_job = dg.define_asset_job(name="my_job", selection="my_asset")

defs = dg.Definitions(assets=[my_first_asset, my_second_asset], jobs=[my_job])
