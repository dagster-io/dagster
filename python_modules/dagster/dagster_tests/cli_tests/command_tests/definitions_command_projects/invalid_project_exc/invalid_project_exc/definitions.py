raise Exception("This is a test exception")
