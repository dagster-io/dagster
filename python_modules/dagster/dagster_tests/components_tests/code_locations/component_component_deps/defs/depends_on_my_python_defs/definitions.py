from pathlib import Path

import dagster as dg
from dagster.components.core.context import ComponentLoadContext


@dg.definitions
def defs(context: ComponentLoadContext) -> dg.Definitions:
    @dg.asset(
        deps=context.component_tree.build_defs_at_path(
            Path("my_python_defs")
        ).resolve_all_asset_specs()
    )
    def downstream_of_all_my_python_defs(): ...

    return dg.Definitions(assets=[downstream_of_all_my_python_defs])
