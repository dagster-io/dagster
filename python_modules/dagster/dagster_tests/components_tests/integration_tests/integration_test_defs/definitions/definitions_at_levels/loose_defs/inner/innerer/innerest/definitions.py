import dagster as dg


@dg.asset
def innerest_defs() -> None: ...


defs = dg.Definitions(assets=[innerest_defs])
