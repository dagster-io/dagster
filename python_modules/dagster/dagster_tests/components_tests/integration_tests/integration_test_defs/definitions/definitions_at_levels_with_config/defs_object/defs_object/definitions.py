import dagster as dg

from .asset import defs_obj_outer  # noqa
from .inner.asset import defs_obj_inner  # noqa

defs = dg.Definitions(assets=[defs_obj_inner, defs_obj_outer])
