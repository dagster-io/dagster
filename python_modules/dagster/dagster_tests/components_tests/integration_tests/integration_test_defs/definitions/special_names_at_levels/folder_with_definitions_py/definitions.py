import dagster as dg


@dg.asset
def asset_in_definitions_py() -> None: ...
