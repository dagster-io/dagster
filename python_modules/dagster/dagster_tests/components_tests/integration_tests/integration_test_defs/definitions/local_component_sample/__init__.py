from dagster._core.definitions.definitions_class import Definitions
from dagster.components import Component, ComponentLoadContext


class MyComponent(Component):
    a_string: str
    an_int: int

    def build_defs(self, context: ComponentLoadContext) -> Definitions:
        return Definitions()
