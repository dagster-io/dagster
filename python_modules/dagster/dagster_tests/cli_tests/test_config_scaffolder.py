from dagster import (
    Int,
    OpDefinition,
    ResourceDefinition,
    _check as check,
)
from dagster._cli.config_scaffolder import scaffold_job_config, scaffold_type
from dagster._config import config_type
from dagster._core.definitions.graph_definition import GraphDefinition


def fail_me():
    return check.failed("Should not call")


def test_scalars():
    assert scaffold_type(config_type.Int()) == 0
    assert scaffold_type(config_type.String()) == ""
    assert scaffold_type(config_type.Bool()) is True
    assert scaffold_type(config_type.Any()) == "AnyType"


def test_basic_ops_config(snapshot):
    job_def = GraphDefinition(
        name="BasicOpsConfigGraph",
        node_defs=[
            OpDefinition(
                name="required_field_op",
                config_schema={"required_int": Int},
                compute_fn=lambda *_args: fail_me(),
            )
        ],
    ).to_job()

    env_config_type = job_def.run_config_schema.config_type

    assert env_config_type.fields["ops"].is_required  # pyright: ignore[reportAttributeAccessIssue]
    ops_config_type = env_config_type.fields["ops"].config_type  # pyright: ignore[reportAttributeAccessIssue]
    assert ops_config_type.fields["required_field_op"].is_required
    required_op_config_type = ops_config_type.fields["required_field_op"].config_type
    assert required_op_config_type.fields["config"].is_required

    assert set(env_config_type.fields["loggers"].config_type.fields.keys()) == set(["console"])  # pyright: ignore[reportAttributeAccessIssue]

    console_logger_config_type = env_config_type.fields["loggers"].config_type.fields["console"]  # pyright: ignore[reportAttributeAccessIssue]

    assert set(console_logger_config_type.config_type.fields.keys()) == set(["config"])

    assert console_logger_config_type.config_type.fields["config"].is_required is False

    console_logger_config_config_type = console_logger_config_type.config_type.fields[
        "config"
    ].config_type

    assert set(console_logger_config_config_type.fields.keys()) == set(["log_level", "name"])

    snapshot.assert_match(scaffold_job_config(job_def, skip_non_required=False))


def dummy_resource(config_field):
    return ResourceDefinition(lambda _: None, config_field)
