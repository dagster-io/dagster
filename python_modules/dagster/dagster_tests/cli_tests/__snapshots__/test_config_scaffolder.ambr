# serializer version: 1
# name: test_basic_ops_config
  dict({
    'execution': dict({
      'config': dict({
        'in_process': dict({
          'marker_to_close': '',
          'retries': dict({
            'disabled': dict({
            }),
            'enabled': dict({
            }),
          }),
          'step_dependency_config': dict({
            'require_upstream_step_success': True,
          }),
        }),
        'multiprocess': dict({
          'max_concurrent': None,
          'retries': dict({
            'disabled': dict({
            }),
            'enabled': dict({
            }),
          }),
          'start_method': dict({
            'forkserver': dict({
              'preload_modules': list([
              ]),
            }),
            'spawn': dict({
            }),
          }),
          'step_dependency_config': dict({
            'require_upstream_step_success': True,
          }),
          'tag_concurrency_limits': list([
          ]),
        }),
      }),
    }),
    'loggers': dict({
      'console': dict({
        'config': dict({
          'log_level': '',
          'name': '',
        }),
      }),
    }),
    'ops': dict({
      'required_field_op': dict({
        'config': dict({
          'required_int': 0,
        }),
      }),
    }),
    'resources': dict({
      'io_manager': dict({
        'config': 'AnyType',
      }),
    }),
  })
# ---
